chore(docs): Updates Title Case and pronouns (#18476)
gatsbyjs/gatsby: docs/docs/creating-a-transformer-plugin.md

`gatsby-transformer-yaml` is transformer plugin that looks for new nodes with a media type of text/yaml (e.g. a .yaml file) and creates new YAML child node(s) by parsing the YAML source into JavaScript objects.

> Checkout this example of rebuilding a simplified `gatsby-transformer-yaml` directly in an example site. For this example, you have aa default gatsby starter site which includes a `src/data/example.yml` file:

Reviewer: ```suggestion
Check out this example of rebuilding a simplified `gatsby-transformer-yaml` directly in a site. Say you have a default Gatsby starter site which includes a `src/data/example.yml` file:
```

Author: This is a great change - thank you! I wasn't sure the best way to simplify it - your suggestion is fantastic.
